### PR TITLE
chore: 🤖 Upgrade `addons/core` to Ember 4.12.2

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: ICU-11341

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-11341)

## Description

Upgrade `addons/core` to Ember 4.12.2

## Checklist:

~~- [ ] I have added before and after screenshots for UI changes~~
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [X] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: addons/core/package.json [hunk 1]
```diff
@@ -45,17 +45,19 @@
...
+    "@ember/test-helpers": "^2.9.3",
     "@embroider/test-setup": "^2.1.1",
     "@faker-js/faker": "^8.0.2",
     "@glimmer/component": "^1.1.2",
     "@glimmer/tracking": "^1.1.2",
     "broccoli-asset-rev": "^3.0.0",
+    "concurrently": "^8.0.1",
     "doctoc": "^2.2.0",
-    "ember-cli": "^4.4.1",
+    "ember-cli": "~4.12.2",
```
[the remark this answers is not in the document]
Comment (ZedLi) [reply addons/core/package.json:60]: Previous upgrades use `minor` didn't they? Why switch to exact?

[the remark this answers is not in the document]
Comment (dvonanderson) [reply addons/core/package.json:60]: @ZedLi you are right, I think it was just a mishap on my behalf, we should be doing minor and point releases.  The cli for ember-upgrade replaces the pre-fix.  I will fix

File: addons/core/.template-lintrc.js [hunk 2]
```diff
@@ -8,4 +8,17 @@
...
+  rules: {
+    'no-curly-component-invocation': {
+      allow: ['app-name', 'company-name', 'company-copyright'],
+    },
+  },
+  overrides: [
+    {
+      files: ['tests/**/*-test.js'],
+      rules: {
+        'no-unused-block-params': false,
```
[the remark this answers is not in the document]
Comment (dvonanderson) [reply addons/core/.template-lintrc.js:20]: @ZedLi yes I can fix these, this is from the `pending-confirmations-test`.

File: addons/core/tests/integration/components/form/authenticate-test.js [hunk 3]
```diff
@@ -29,10 +29,11 @@ module('Integration | Component | form | authenticate', function (hooks) {
     this.submit = () => {};
     this.disabled = () => {};
 
-    await render(hbs`<Form::Authenticate   
-    @model={{this.model}}
-    @submit={{this.submit}}
-    @disabled={{this.disabled}}/>`);
+    await render(hbs`<Form::Authenticate
+  @model={{this.model}}
+  @onSubmit={{this.submit}}
```
[the remark this answers is not in the document]
Comment (dvonanderson) [reply addons/core/tests/integration/components/form/authenticate-test.js:34]: @ZedLi this is another template-lint error that I will fix for `no-passed-in-event-handlers`

File: addons/core/.template-lintrc.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (calcaide) [reply addons/core/.template-lintrc.js:20]: I agree with @ZedLi comment, we should fix this before adding an override.

File: addons/core/tests/integration/components/image-test.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ module('Integration | Component | Image', function (hooks) {
 
   test('it renders', async function (assert) {
     assert.expect(1);
-    await render(hbs`<Image @name="logo-app" />`);
+    await render(hbs`<Image @name='logo-app' />`);
```
Comment (calcaide) [inline addons/core/tests/integration/components/image-test.js:16]: I have noticed we are changing from double quotes (`"`) to single quotes (`'`) within the test files. What is the reason behind this change? i.e on Admin UI we use double quotes.

File: addons/core/tests/integration/components/form/authenticate-test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (calcaide) [reply addons/core/tests/integration/components/form/authenticate-test.js:34]: I agree with @ZedLi comment, this needs be fixed.

File: addons/core/tests/integration/helpers/reverse-indexed-display-name-test.js [hunk 5]
```diff
@@ -17,26 +17,26 @@ module('Integration | Helper | reverse-indexed-display-name', function (hooks) {
     this.set('array', ['Boundary', 'is', 'cool!']);
 
     await render(hbs`{{reverse-indexed-display-name
-      'resources.session-recording.connection.title_index'
-      this.array
-      'Boundary'
-    }}`);
+  'resources.session-recording.connection.title_index'
```
Comment (calcaide) [inline addons/core/tests/integration/helpers/reverse-indexed-display-name-test.js:20]: I have noticed we changed indentation in test files, but I do not see any changes on prettier config file, what is the reason behind this change?

**Comment by calcaide** [comment]
@dvonanderson 
The PR is to update `addons/core` but I notice changes within rose, admin and desktop. Are those changes necessary for the successful update of `addons/core`? Can we separate those changes within another PR?

File: addons/core/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (calcaide) [reply addons/core/package.json]: Within `addons/auth` which has been recently upgraded, the declaration of `ember-cli` is as follows: "ember-cli": "^4.12.2", It will be good to keep consistency.

**Review by calcaide — CHANGES_REQUESTED** [review]
@dvonanderson 
Nice work on this upgrade 💪 I left some comments/request changes 🙌

File: addons/core/tests/integration/components/image-test.js (hunk 4 above)
Comment (dvonanderson) [reply addons/core/tests/integration/components/image-test.js:16]: @calcaide looking into this.

File: addons/core/tests/integration/helpers/reverse-indexed-display-name-test.js (hunk 5 above)
Comment (dvonanderson) [reply addons/core/tests/integration/helpers/reverse-indexed-display-name-test.js:20]: @calcaide looking into this.

**Comment by dvonanderson** [comment]
cleaning up lint errors and unneeded commits
